### PR TITLE
Simplify symlog range determination logic

### PR DESCRIPTION
**Pull request by dstansby**
This is just a cleanup to logic, and hopefully will form part of a larger attempt to clean and fix various parts of `SymLog` locating and ticking.

### CODE REVIEW

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -2581,52 +2581,33 @@ def tick_values(self, vmin, vmax):
...
+        has_b = (has_a and vmax > -linthresh) or (has_c and vmin < linthresh)
 
         def get_log_range(lo, hi):
             lo = np.floor(np.log(lo) / np.log(base))
             hi = np.ceil(np.log(hi) / np.log(base))
             return lo, hi
 
         # Calculate all the ranges, so we can determine striding
+        a_range = (0, 0)
+        a_upper_lim = np.min([-linthresh, vmax])
```
Comment (timhoffm) [inline lib/matplotlib/ticker.py]: Can be moved into the ` if`.

File: lib/matplotlib/ticker.py [hunk 2]
```diff
@@ -2581,52 +2581,33 @@ def tick_values(self, vmin, vmax):
...
-            if has_b:
-                a_range = get_log_range(linthresh, np.abs(vmin) + 1)
-            else:
-                a_range = get_log_range(np.abs(vmax), np.abs(vmin) + 1)
-        else:
-            a_range = (0, 0)
+            a_range = get_log_range(np.abs(a_upper_lim), np.abs(vmin) + 1)
 
+        c_range = (0, 0)
+        c_lower_lim = np.max([linthresh, vmin])
```
Comment (timhoffm) [inline lib/matplotlib/ticker.py]: Can be moved into the ` if`.

File: lib/matplotlib/ticker.py [hunk 3]
```diff
@@ -2581,52 +2581,33 @@ def tick_values(self, vmin, vmax):
...
+        a_range = (0, 0)
+        a_upper_lim = np.min([-linthresh, vmax])
         if has_a:
-            if has_b:
-                a_range = get_log_range(linthresh, np.abs(vmin) + 1)
-            else:
-                a_range = get_log_range(np.abs(vmax), np.abs(vmin) + 1)
-        else:
-            a_range = (0, 0)
+            a_range = get_log_range(np.abs(a_upper_lim), np.abs(vmin) + 1)
```
Comment (anntzer) [inline lib/matplotlib/ticker.py]: would just directly unpack these into a_lo, a_hi = ... which makes later code a bit more readable (`total_ticks = (a_hi - a_lo) + (c_hi - c_lo)`) imo.

File: lib/matplotlib/ticker.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/ticker.py]: and also that's just `min(-linthresh, vmax)`

**Pull request merged**